### PR TITLE
Add @name option to account settings.

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* Extract account settings in a separate file + extension.
* Add `ChangeHandleViewController` which manages changing a users handle, as this controller has some custom logic which did not fit into the reusable settings approach the common parts with the `SettingsTableViewController` have been extracted into a common super class called `SettingsBaseTableViewController` (e.g. showing the top separator, cell height etc.).
* This updates the `AddEmailPasswordViewController` and the `AddPhoneNumberViewController` to use the new sync-engine API to request and observe updates to the self user.
* Add a developer option to show the handle settings menu (Settings has to be closed and reopened for this to take effect).

---

- [x] This PR depends on a new release of `wire-ios-sync-engine` after the `feature/usernames` branch has been merged.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Registration/AddEmailPasswordViewController.m [hunk 1]
```diff
@@ -184,7 +187,8 @@ - (void)didCompleteFormStep:(UIViewController *)viewController
 - (void)emailVerificationStepDidRequestVerificationEmail
 {
     [self.analyticsTracker tagResentEmailVerification];
-    [[ZMUserSession sharedSession] requestVerificationEmailForEmailUpdate:self.credentials];
+    // TODO: Check error
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Registration/AddEmailPasswordViewController.m]: 😱

File: Wire-iOS/Sources/UserInterface/Registration/AddEmailPasswordViewController.m [hunk 2]
```diff
@@ -172,8 +174,9 @@ - (void)didCompleteFormStep:(UIViewController *)viewController
         
         self.credentials = [ZMEmailCredentials credentialsWithEmail:addEmailStepViewController.emailAddress
                                                            password:addEmailStepViewController.password];
-        
-        [[ZMUserSession sharedSession] requestVerificationEmailForEmailUpdate:self.credentials];
+
+        // TODO: Check error
```
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Registration/AddEmailPasswordViewController.m]: 😱

**Comment by marcoconti83** [comment]
wire-sync-engine available as 60.2.0

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -761,6 +761,9 @@
 		BF78687F1D86B3710063762F /* LocationDataTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF78687D1D86B36D0063762F /* LocationDataTests.swift */; };
 		BF7EF5451D953B3F003E3999 /* Date+Format.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF7EF5441D953B3F003E3999 /* Date+Format.swift */; };
 		BF8042271BFC741F004CD789 /* NSURL+WireURLs.m in Sources */ = {isa = PBXBuildFile; fileRef = BF8042261BFC741F004CD789 /* NSURL+WireURLs.m */; };
+		BF808B4D1DE5AD1600718076 /* SettingsCellDescriptorFactory+Account.swift in Sources */ = {isa = PBXBuildFile; fileRef = BF808B4C1DE5AD1600718076 /* SettingsCellDescriptorFactory+Account.swift */; };
+		BF808B4F1DE5C25500718076 /* ChangeHandleViewController .swift in Sources */ = {isa = PBXBuildFile; fileRef = BF808B4E1DE5C25500718076 /* ChangeHandleViewController .swift */; };
```
Comment (mikeger) [inline Wire-iOS.xcodeproj/project.pbxproj]: This file name contains space ❗️

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    }
+
+    func handleElement() -> SettingsCellDescriptorType {
+        if nil != ZMUser.selfUser().handle {
+            return SettingsExternalScreenCellDescriptor(
+                title: "self.settings.account_section.handle.title".localized,
+                isDestructive: false,
+                presentationStyle: .navigation,
+                presentationAction: ChangeHandleViewController.init,
+                previewGenerator: { _ in .text("@\(ZMUser.selfUser().handle ?? "")") },
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift]: Can be potentially done with "@" + ZMUser.selfUser().handle ?? "", then code is somewhat cleaner (just an opinion)

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/ChangeHandleViewController .swift [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+        performWiggleAnimation()
+        return false
+    }
+}
+
+/// This struct represents the current state of a handle
+/// change operation and performs necessary validation steps of
+/// a new handle. The `ChangeHandleViewController` uses this state 
+/// to layout its interface.
+struct HandleChangeState {
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/ChangeHandleViewController .swift]: Nice stuff!

File: Wire-iOS.xcodeproj/project.pbxproj (hunk 3 above)
Comment (daehn) [inline Wire-iOS.xcodeproj/project.pbxproj]: 😱

File: Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift (hunk 4 above)
Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift]: There was a bug there anyways, I think we do not want to display the @ sign if the user does not yet have a handle, fixed now.

**Comment by daehn** [comment]
@mikeger please re-review

**Pull request merged**